test/integration/controller/jobs/pod/pod_controller_test.go

```diff
@@ -513,6 +513,81 @@ var _ = ginkgo.Describe("Pod controller interacting with scheduler", ginkgo.Orde
 		util.ExpectReservingActiveWorkloadsMetric(clusterQueue, 1)
 	})
 
+	ginkgo.It("Should schedule pod groups as they fit in their ClusterQueue", func() {
+		ginkgo.By("creating localQueue")
+		localQueue = testing.MakeLocalQueue("local-queue", ns.Name).ClusterQueue(clusterQueue.Name).Obj()
+		gomega.Expect(k8sClient.Create(ctx, localQueue)).Should(gomega.Succeed())
+
+		pod1 := testingpod.MakePod("dev-pod1", ns.Name).
```

Comment:
Let's test with 2 different roles, 2 pods each.

Reply:
Changed the existing test in https://github.com/kubernetes-sigs/kueue/pull/1319/commits/fb5a4eb591521e5e3e2d7acfb4b6268a697b52e9